.gitpod.yml

```
@@ -1 +1,77 @@
-mainConfiguration: https://github.com/SU-SWS/acsf-cardinalsites-public
+github:
+  prebuilds:
+    master: true
+    branches: false
+    pullRequests: true
+    pullRequestsFromForks: false
+    addCheck: false
+    addComment: false
+    addBadge: false
+ports:
+  - port: 3306
+    onOpen: ignore
+    visibility: private
+  - port: 33060
+    onOpen: ignore
+    visibility: private
+  - port: 8001
+    onOpen: ignore
+    visibility: public
+image: pookmish/drupal8ci:gitpod
+checkoutLocation: soe_profile
+workspaceLocation: /workspace/stack
+tasks:
+  - before: >
+      rm -rf ~/.ssh/id_rsa &&
+      rm -rf ~/.ssh/id_rsa.pub &&
+      eval $(command gp env -e) &&
+      mkdir -p ~/.ssh &&
+      [[ ! -z $SSH_PUBLIC_KEY  ]] &&
+      echo $SSH_PUBLIC_KEY | base64 -d > ~/.ssh/id_rsa.pub &&
+      chmod 644 ~/.ssh/id_rsa.pub &&
+      [[ ! -z $SSH_PRIVATE_KEY  ]] &&
+      echo $SSH_PRIVATE_KEY | base64 -d > ~/.ssh/id_rsa &&
+      chmod 600 ~/.ssh/id_rsa &&
+      [[ ! -z $GITCONFIG  ]] &&
+      echo $GITCONFIG | base64 -d > ~/.gitconfig &&
+      chmod 644 ~/.gitconfig
+    init: >
```

Should this be: 
```suggestion
  - init: >
```
?  

The docs seem to indicate that `before` `init` and `coomand` are the same sort of thing, and it seems like they should have the same yml?

---

Maybe same with `command` below?

---

Good question, but no. Basically the `tasks` is an array of arrays. Look at it in the way of a php array
```
'tasks' => [
  [
    'before' => ...
    'init' => ...
    'command' => ...
  ],
  [
    'before' => ...
    'command' => ...
  ],
]
```
those three are treated as the same task. adding the `-` would then treat them as separate tasks. see the first example here: https://www.gitpod.io/docs/references/gitpod-yml#tasks
> Each array element opens in its own terminal.

We could do things in parallel like initiate drupal and spin up a decoupled server directory at the same time with 2 different terminals. But we want things to happen sequentially for this implementation.

---

Gotcha.  thanks for the clarification.